frontend/src/components/EventModal.jsx

```diff
@@ -0,0 +1,45 @@
+import { useParams, useNavigate } from "react-router";
+import "../styles/EventModal.css";
+import { useQuery } from "@tanstack/react-query";
+import axios from "axios";
+import SingleEventInfo from "./SingleEventInfo";
+
+const EventModal = () => {
+    const {eventID} = useParams()
+    const navigate = useNavigate()
+
+    const getEvent = useQuery({
+        queryKey: ['event', eventID],
+        queryFn: () => {
+            const url = import.meta.env.VITE_SERVER_API;
+            return axios.get(`${url}/events/${eventID}`, {
+                withCredentials: true,
+            });
+        },
+        refetchOnWindowFocus: false,
+    });
+
+    if (getEvent.isPending) {
+        return (
+            <div className="modal-overlay">
+            <aside className="modal-content">
+                <h4>Loading...</h4>
+            </aside>
+        </div>
+        )
+    }
+
+    if (getEvent.isError) {
+        navigate('/feed')
```

Comment:
This can be done later on (maybe it is) but would you want to show an error if there's an error?